revenue_app/templates/revenue_app/query.html

```diff
@@ -64,9 +64,25 @@ <h1>Queries</h1>
   {% csrf_token %}
   <div class="row mt-3">
     <div class="col-12">
-      <button type="submit" class="btn btn-block btn-evb-orange">Make Query</button>
+      <button id="submit-btn" type="submit" class="btn btn-block btn-evb-orange">Make Query</button>
     </div>
   </div>
 </form>
+<div style="display: flex; justify-content: center;">
+    <h6 id="gif"style="display: none">
+        <img src="{% static 'busy.gif' %}" height="42" width="42" >
+        Running queries...
+    </h6>
+</div>
 <br>
 {% endblock content %}
+{% block scripts %}
+<script type="text/javascript">
+	$(document).ready(function () {
+		$("#query-form").submit(function () {
+			$("#submit-btn").attr("disabled", true);
+			$('#gif').show();
```

Comment:
question: When do you hide the gif? The entire page is reloaded?

Reply:
+1

Reply:
Yes, the entire page is reloaded

Reply:
The gif is hidden, when the user press the button "Make queries" the gif shows up until the whole is page is reloaded again when the queries finished (So, the gif is again hidden).

Reply:
Why is the entire page reloaded when the query finishes?

Reply:
Because the page is render again with error messages to tell the user what the error is or success messages in case the query finishes successfully.